sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -146,12 +153,15 @@ export class JsonRpcProvider extends Provider {
     //   'Client-Target-Api-Version': TARGETED_RPC_VERSION,
     // });
     // TODO: add header for websocket request
-    this.client = new JsonRpcClient(this.connection.fullnode);
-    this.wsClient = new WebsocketClient(
-      this.connection.websocket,
-      opts.skipDataValidation!,
-      opts.socketOptions,
-    );
+    this.client = opts.rpcClient ?? new JsonRpcClient(this.connection.fullnode);
```

Comment:
Cool! In a separate PR, I will move logic on line 151-153 back to client.ts to avoid the version info from being overridden

Reply:
Oh I can shuffle the header config over while I'm here too.

Reply:
Shuffled this around a bit